test/integration/style-spec/validate_spec.test.ts

```diff
@@ -19,10 +19,80 @@ describe('validate_spec', () => {
     });
 
     test('errors from validate do not contain line numbers', () => {
-        const style = JSON.parse(fs.readFileSync('test/integration/style-spec/tests/bad-color.input.json', 'utf8'));
+        const style = JSON.parse(
+            fs.readFileSync(
+                'test/integration/style-spec/tests/bad-color.input.json',
+                'utf8'
+            )
+        );
 
         const result = validate(style, reference);
         expect(result[0].line).toBeUndefined();
     });
 
+    test('Validate sdk-support in spec', () => {
+        const issueTrackers = {
+            js: 'https://github.com/maplibre/maplibre-gl-js/issues',
+            android: 'https://github.com/maplibre/maplibre-native/issues',
+            ios: 'https://github.com/maplibre/maplibre-native/issues',
+        };
+        const platforms = Object.keys(issueTrackers);
+
+        function validatePlatforms(platformSupport, path) {
+            const notSupportedOnAnyPlatform = platforms.every(
+                (platform) => !platformSupport[platform]
+            );
+
+            for (const platform of platforms) {
+                if (notSupportedOnAnyPlatform) {
+                    // don't consider it a problem is something is not supported on any platform
+                    return;
+                }
+                if (!platformSupport[platform])
+                    throw new Error(`Missing platform '${platform}' in sdk-support at ${path.join(
+                        '.'
+                    )}.
+Please create a tracking issue and add the link.`);
+
+                const maplibreIssue =
+          /https:\/\/github.com\/maplibre\/[^/]+\/issues\/(\d+)/;
+                const version = /^\d+\.\d+\.\d+$/;
+                const values = new Set(['supported', 'wontfix']);
+
+                const support = platformSupport[platform];
+                if (
+                    !support.match(maplibreIssue) &&
+          !support.match(version) &&
+          !values.has(support)
+                ) {
+                    throw new Error(
```

Comment:
This is not how tests should work - you should have `expect(somethign).toBe(something elese).
In this case I would recommend something like:
```ts
expect(support.match(maplibreIssue).toBeFalsy();
expect(support.match(version)).toBeFalsy();
expect(values.has(support)).toBeFalsy();
```

Reply:
That is not equivalant though, and there should be a helpful message when it fails.

Reply:
You can have the test name to be meaningful, so that when it fails it helps understanding what went wrong. 
What I'm concerned about is that lint and other tools may report that the test doesn't have an expect, which isn't great in terms of future proofing this.